### PR TITLE
chore(deps): Update to Node 16

### PR DESCRIPTION
**Pull request by akash1810**
This is a retake of https://github.com/guardian/actions-read-private-repos/pull/6 to test the assumption that forks don't build properly.

## What does this change?
The Node 12 runtime is no longer an LTS version, and GitHub have also started [marking it as deprecated](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/).

Update to Node 16, which also brings a new version to the lockfile, v2.

It doesn't look like Node 20 is supported by GitHub yet.

## How to test
See CI.

## How can we measure success?
Fewer warnings in builds of projects that use this action.

### CODE REVIEW

**Review by michel-ds — APPROVED** [review]
LGTM but I'm not a domain expert. Thanks for bumping.

File: package.json
Comment (michel-ds) [inline package.json:1]: Does the version need bumping?
https://github.com/guardian/actions-read-private-repos/blob/8604c0bff6d245ba0e66e1cc471877818a12dbed/package.json#L3

Comment (akash1810) [reply package.json:1]: Ah, yep. Will do this separately for ease as git tags also need updating.

**Comment by akash1810** [comment]
This is available in [v0.1.1](https://github.com/guardian/actions-read-private-repos/releases/tag/v0.1.1) now.

**Pull request merged**